src/main/java/org/rulez/demokracia/PDEngine/VoteRegistry.java

```diff
@@ -28,4 +36,97 @@ public static Vote getByKey(String adminKey) {
 		return votes.get(adminKey);
 	}
 
+    public static void checkVoteName(String voteName) throws ReportedException {
+
+
+		ErrorReason checkValue = checkString(voteName);
+
+		if (checkValue == ErrorReason.STR_SHORT) {
+            throw new ReportedException("Vote name is too short!");
+		} else if (checkValue == ErrorReason.STR_LONG) {
+            throw new ReportedException("Vote name is too long!");
+		} else if (checkValue ==ErrorReason.CHAR_NOT_VALID ) {
+            throw new ReportedException("Wrong characters in the vote name!");
+		}
+
+
+    }
+
+    public static void checkNeededAssurances (List<String> assurance) throws ReportedException {
+
+		ErrorReason checkValue = checkAssurances(assurance);
+
+		if (checkValue == ErrorReason.STR_SHORT) {
+            throw new ReportedException("neededAssurance is too short!");
+		} else if (checkValue == ErrorReason.STR_LONG) {
+            throw new ReportedException("neededAssurance is too long!");
+		} else if (checkValue == ErrorReason.CHAR_NOT_VALID) {
+            throw new ReportedException("Wrong characters in the neededAssurance!");
+        } else if (checkValue == ErrorReason.DUPLICATE_VALUE) {
+            throw new ReportedException("neededAssurance contains duplicated values!");
+        }
+	}
+
+    public static void checkCountedAssurances (List<String> assurance) throws ReportedException {
+
+        ErrorReason checkValue = checkAssurances(assurance);
+
+        if (checkValue == ErrorReason.STR_SHORT) {
+            throw new ReportedException("countedAssurance is too short!");
+        } else if (checkValue == ErrorReason.STR_LONG) {
+            throw new ReportedException("countedAssurance is too long!");
+        } else if (checkValue == ErrorReason.CHAR_NOT_VALID) {
+            throw new ReportedException("Wrong characters in the countedAssurance!");
+        } else if (checkValue == ErrorReason.DUPLICATE_VALUE) {
+            throw new ReportedException("countedAssurance contains duplicated values!");
+        }
+    }
+
+	private static ErrorReason checkAssurances(List<String> list) throws ReportedException{
+
+		//Set<T> duplicates = new LinkedHashSet<T>();
+
+		Set<String> uniques = new HashSet<>();
+		ErrorReason ret = ErrorReason.READY;
```

Comment:
just throw an exception when there is a problem. it is okay to return void if all is okay.